core/rawdb/accessors_indexes.go

```diff
@@ -27,28 +29,34 @@ import (
 
 // ReadTxLookupEntry retrieves the positional metadata associated with a transaction
 // hash to allow retrieving the transaction or receipt by hash.
-func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) common.Hash {
+func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) *uint64 {
 	data, _ := db.Get(txLookupKey(hash))
 	if len(data) == 0 {
-		return common.Hash{}
+		return nil
+	}
+	// Database v6 tx lookup just stores the block number
+	if len(data) < common.HashLength {
```

Comment:
I am thinking is this the final step of `txLookup`? Since there is no flag at all now for the v6 format.

Reply:
Writing was updated to use this format: https://github.com/ethereum/go-ethereum/pull/19431/files#diff-0fba94aa54693ff2a4f26210c404dbd9R59

Reply:
Not clear what you mean by a flag.

Reply:
I mean the "flag" for blob. For example, if the blob can be decoded into v3 structure, this is kind of "flag". Also if the length of blob is 32, it is kind of flag for v4, v5.

Reply:
Well, the flag here is `< 32 bytes` :) But all in all I think we always want to try to decode into the newest version first. People who upgrade to 1.9 will with a high chance resync, so might as well use the happy path and not do 3 decoding to get it right.

Reply:
@rjl493456442 My intention was how @karalabe interpreted the code. If either of you feel like this needs to be made more clear and have a suggestion about how to do it please feel free to adjust it or let me know what to do.